pkg/gardenlet/controller/shootsecret/reconciler.go

```diff
@@ -26,42 +27,40 @@ import (
 	"github.com/gardener/gardener/pkg/extensions"
 	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
 
-	"github.com/sirupsen/logrus"
+	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 const finalizerName = "gardenlet.gardener.cloud/secret-controller"
 
 type reconciler struct {
-	log          logrus.FieldLogger
 	gardenClient client.Client
 	seedClient   client.Client
 }
 
 // NewReconciler returns a new reconciler for secrets related to shoots.
-func NewReconciler(gardenClient, seedClient client.Client, log logrus.FieldLogger) reconcile.Reconciler {
+func NewReconciler(gardenClient, seedClient client.Client) reconcile.Reconciler {
 	return &reconciler{
-		log:          log,
 		gardenClient: gardenClient,
 		seedClient:   seedClient,
 	}
 }
 
 func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	log := r.log.WithField("secret", request)
+	log := logf.FromContext(ctx)
 
 	secret := &corev1.Secret{}
 	if err := r.seedClient.Get(ctx, request.NamespacedName, secret); err != nil {
 		if apierrors.IsNotFound(err) {
-			log.Debug("Skipping because Secret does not exist anymore")
+			log.Info("Object is gone, stop reconciling")
```

Comment:
Nit: Shouldn't a former debug message have `V` level > 0?
```suggestion
			log.V(1).Info("Object is gone, stop reconciling")
```

Reply:
For now, let's keep it like this since this is also what is done in all other controllers that were refactored earlier. We might decide to move this to `V(1)` eventually (but then let's do it consistently across all controllers) :)